File: totosuki/ABC/B/021.py
_ = input()
a, b = map(int, input().split())
_ = input()
P = list(map(int, input().split())) + [a,b]
if len(P) == len(set(P)):
  print("YES")
else:
  print("NO")

N = int(input())
a, b = map(int, input().split())
K = int(input())
P = list(map(int, input().split()))
l = P + [a,b]
if len(l) == len(set(l)):
  print("YES")
else:
  print("NO")

File: OCH_Tao/ABC/A/021.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const binN = N.toString(2);
const binNList = [...binN];
const NDigit = binNList.length;
let count = 0;
for (let i = 0; i < NDigit; i++) {
  if (binNList[i] === "1") {
    count++;
  }
}
console.log(count);
for (let i = NDigit; i > 0; i--) {
  if (binNList[i - 1] === "1") {
    console.log(2 ** (NDigit - i));
  }
}